src/components/gridList/gridList.js

```diff
@@ -747,8 +760,10 @@ function GridTileDirective($mdMedia) {
     if (angular.isDefined(scope.$parent.$index)) {
       scope.$watch(function() { return scope.$parent.$index; },
         function indexChanged(newIdx, oldIdx) {
-          gridCtrl.removeTile(element, attrs);
-          gridCtrl.addTile(element, attrs, newIdx);
+          if (newIdx === oldIdx) {
+            return;
+          }
+          gridCtrl.reorderTiles();
```

Comment:
yes, it is computation intensive, I didn't have time to go deeper into the codebase at the time, will probably try to find a better solution today :)